compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1238,7 +1238,7 @@ class Typer extends Namer
           args = args.take(tparams.length)
```

Comment:
The test failure seems genuinely to appear with this new commit (not with my changes) — everything works with the parent.

```
sbt:dotty> dotc -d out -Ycheck:all tests/pos/t1786-cycle.scala -Yforce-sbt-phases
[warn] Multiple main classes detected.  Run 'show discoveredMainClasses' to see the list
[info] Running (fork) dotty.tools.dotc.Main -classpath /Users/pgiarrusso/git/dotty/library/../out/bootstrap/dotty-library-bootstrapped/scala-0.10/dotty-library_0.10-0.10.0-bin-SNAPSHOT.jar -d out -Ycheck:all tests/pos/t1786-cycle.scala -Yforce-sbt-phases
checking tests/pos/t1786-cycle.scala after phase frontend
exception occurred while compiling tests/pos/t1786-cycle.scala
Exception in thread "main" dotty.tools.dotc.core.TypeError: bad parameter reference LongTraversableLike#Repr at sbt-deps
the parameter is type Repr in trait LongTraversableLike but the prefix LongTraversableLike
does not define any corresponding arguments.
	at dotty.tools.dotc.core.Types$NamedType.argDenot(Types.scala:1834)
	at dotty.tools.dotc.core.Types$NamedType.fromDesignator$1(Types.scala:1751)
	at dotty.tools.dotc.core.Types$NamedType.computeDenot(Types.scala:1769)
	at dotty.tools.dotc.core.Types$NamedType.denot(Types.scala:1725)
	at dotty.tools.dotc.core.Types$NamedType.info(Types.scala:1714)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector$TypeDependencyTraverser.traverse(ExtractDependencies.scala:422)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4485)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4483)
	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:4469)
	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:4486)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector$TypeDependencyTraverser.traverse(ExtractDependencies.scala:432)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4485)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4483)
	at dotty.tools.dotc.core.Types$TypeAccumulator.$anonfun$foldOver$5(Types.scala:4405)
	at dotty.tools.dotc.core.Types$VariantTraversal.atVariance(Types.scala:3942)
	at dotty.tools.dotc.core.Types$TypeAccumulator.foldArgs$1(Types.scala:4405)
	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:4409)
	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:4486)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector$TypeDependencyTraverser.traverse(ExtractDependencies.scala:432)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4485)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4483)
	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:4435)
	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:4486)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector$TypeDependencyTraverser.traverse(ExtractDependencies.scala:432)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector$TypeDependencyTraverser.traverse(ExtractDependencies.scala:422)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4485)
	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:4483)
	at dotty.tools.dotc.core.Types$TypeAccumulator.$anonfun$foldOver$5(Types.scala:4405)
	at dotty.tools.dotc.core.Types$VariantTraversal.atVariance(Types.scala:3942)
	at dotty.tools.dotc.core.Types$TypeAccumulator.foldArgs$1(Types.scala:4405)
	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:4409)
	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:4486)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector$TypeDependencyTraverser.traverse(ExtractDependencies.scala:432)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.addTypeDependency(ExtractDependencies.scala:441)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.traverse(ExtractDependencies.scala:354)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1413)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1321)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1416)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.traverse(ExtractDependencies.scala:373)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1413)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1335)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1416)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.traverse(ExtractDependencies.scala:373)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1413)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.$anonfun$apply$1(Trees.scala:1297)
	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:122)
	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:118)
	at scala.collection.immutable.List.foldLeft(List.scala:86)
	at scala.collection.TraversableOnce.$div$colon(TraversableOnce.scala:151)
	at scala.collection.TraversableOnce.$div$colon$(TraversableOnce.scala:151)
	at scala.collection.AbstractTraversable.$div$colon(Traversable.scala:104)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1297)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1333)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1416)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.traverse(ExtractDependencies.scala:373)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.$anonfun$traverse$3(ExtractDependencies.scala:371)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.$anonfun$traverse$3$adapted(ExtractDependencies.scala:371)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.traverse(ExtractDependencies.scala:371)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1413)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1377)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1416)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.traverse(ExtractDependencies.scala:373)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1415)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1413)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.$anonfun$apply$1(Trees.scala:1297)
	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:122)
	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:118)
	at scala.collection.immutable.List.foldLeft(List.scala:86)
	at scala.collection.TraversableOnce.$div$colon(TraversableOnce.scala:151)
	at scala.collection.TraversableOnce.$div$colon$(TraversableOnce.scala:151)
	at scala.collection.AbstractTraversable.$div$colon(Traversable.scala:104)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1297)
	at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1383)
	at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1416)
	at dotty.tools.dotc.sbt.ExtractDependenciesCollector.traverse(ExtractDependencies.scala:373)
	at dotty.tools.dotc.sbt.ExtractDependencies.run(ExtractDependencies.scala:65)
	at dotty.tools.dotc.core.Phases$Phase.$anonfun$runOn$1(Phases.scala:298)
	at scala.collection.immutable.List.map(List.scala:283)
	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:296)
	at dotty.tools.dotc.core.Phases$Phase.runOn$(Phases.scala:295)
	at dotty.tools.dotc.sbt.ExtractDependencies.runOn(ExtractDependencies.scala:47)
	at dotty.tools.dotc.Run.$anonfun$compileUnits$3(Run.scala:175)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.trackTime(Stats.scala:49)
	at dotty.tools.dotc.Run.$anonfun$compileUnits$2(Run.scala:172)
	at dotty.tools.dotc.Run.$anonfun$compileUnits$2$adapted(Run.scala:170)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:194)
	at dotty.tools.dotc.Run.runPhases$1(Run.scala:170)
	at dotty.tools.dotc.Run.$anonfun$compileUnits$1(Run.scala:195)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:90)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:150)
	at dotty.tools.dotc.Run.compileSources(Run.scala:137)
	at dotty.tools.dotc.Run.compile(Run.scala:121)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:31)
	at dotty.tools.dotc.Driver.process(Driver.scala:134)
	at dotty.tools.dotc.Driver.process(Driver.scala:103)
	at dotty.tools.dotc.Driver.process(Driver.scala:115)
	at dotty.tools.dotc.Driver.main(Driver.scala:142)
	at dotty.tools.dotc.Main.main(Main.scala)
```

Reply:
This is fixed in the latest commit.